node-config-docsgen/action.yml

```diff
@@ -0,0 +1,159 @@
+      id: generate_docs
+      shell: bash
+      working-directory: ./stacks-core
+      env:
+        OUTPUT_FILE: ${{ inputs.output }}
+        PROJECT_ROOT: $(pwd)
+      run: |
+        echo "Generating configuration documentation..."
+
+        # Navigate to the script directory and run
+        chmod +x contrib/tools/config-docs-generator/generate-config-docs.sh
+        bash contrib/tools/config-docs-generator/generate-config-docs.sh
+
+        # Verify output was generated
+        if [[ ! -f "$OUTPUT_FILE" ]]; then
+          echo "Error: Configuration documentation was not generated"
+          echo "Expected file: $OUTPUT_FILE"
+          exit 1
+        fi
+
+        echo "Documentation generated successfully at $OUTPUT_FILE"
+
+    ## Validate generated documentation quality
+    - name: Validate Generated Documentation
+      id: validate_docs
+      shell: bash
+      working-directory: ./stacks-core
+      run: |
+        echo "Validating generated documentation..."
+
+        OUTPUT_FILE="${{ inputs.output }}"
+
+        # Check if file exists and has content
+        if [[ ! -f "$OUTPUT_FILE" ]]; then
+          echo "Error: Markdown file not found at $OUTPUT_FILE"
+          exit 1
+        fi
+
+        # Check file size (should be substantial)
+        FILE_SIZE=$(wc -c < "$OUTPUT_FILE")
+        MIN_SIZE=${{ inputs.min-doc-size }}
+
+        if [[ $FILE_SIZE -lt $MIN_SIZE ]]; then
+          echo "Error: Generated documentation is too small ($FILE_SIZE bytes, minimum $MIN_SIZE)"
+          echo "This likely indicates a generation failure"
+          exit 1
+        fi
+
+        # Check word count and basic structure
+        WORD_COUNT=$(wc -w < "$OUTPUT_FILE")
+        LINE_COUNT=$(wc -l < "$OUTPUT_FILE")
+
+        echo "Documentation validation results:"
+        echo "  - File size: $FILE_SIZE bytes"
+        echo "  - Word count: $WORD_COUNT words"
+        echo "  - Line count: $LINE_COUNT lines"
+
+        # Basic content validation
+        if ! grep -q "# Configuration Reference" "$OUTPUT_FILE"; then
+          echo "Warning: Documentation may be malformed - missing main title"
+        fi
+
+        if [[ $WORD_COUNT -lt 1000 ]]; then
+          echo "Warning: Documentation seems very short ($WORD_COUNT words)"
+        fi
+
+        echo "Documentation validation completed successfully ✅"
+
+    ## Upload documentation as artifact
+    - name: Upload Documentation Artifact
+      id: upload_artifact
+      uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+      working-directory: ./stacks-core
+      with:
+        name: node-parameters-bundle
+        path: ${{ inputs.output }}
+        retention-days: 30
+        if-no-files-found: error
+
+    ## Generate job summary
+    - name: Generate Job Summary
+      id: summary
+      shell: bash
+      working-directory: ./stacks-core
+      run: |
+        echo "## Configuration Documentation Generated ✅" >> $GITHUB_STEP_SUMMARY
```

Comment:
if we do collect the error/warn - we'd want to adjust this step a little (a green check implies it passed, and the warns above suggest that may not be true). 
a simple if/else may work, or a generic adjustment or the text in this step that doesn't infer the text is passing or failing - it's just the output from previous steps and some data about the output.

Reply:
I don't think this is relevant anymore. if the generated doc does not produce a file size greater that 50000 byes (from memory, i think that's what it was),  or other errors are encountered - the workflow will fail and this step will not run.